lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -485,21 +486,36 @@ static int z_to_nrf_protocol(int proto)
 	}
 }
 
-static int nrf_to_z_dns_error_code(int nrf_error)
+static int nrf_to_z_dns_error_code(int nrf_gai_err)
 {
-	switch (nrf_error) {
-	case NRF_ENOMEM:
-		return DNS_EAI_MEMORY;
-	case NRF_EAGAIN:
+	switch (nrf_gai_err) {
+	case NRF_EAI_ADDRFAMILY:
+		return DNS_EAI_ADDRFAMILY;
+	case NRF_EAI_AGAIN:
 		return DNS_EAI_AGAIN;
-	case NRF_EAFNOSUPPORT:
+	case NRF_EAI_BADFLAGS:
+		return DNS_EAI_BADFLAGS;
+	case NRF_EAI_FAIL:
+		return DNS_EAI_FAIL;
+	case NRF_EAI_FAMILY:
+		return DNS_EAI_FAMILY;
+	case NRF_EAI_MEMORY:
+		return DNS_EAI_MEMORY;
+	case NRF_EAI_NODATA:
+		return DNS_EAI_NODATA;
+	case NRF_EAI_NONAME:
 		return DNS_EAI_NONAME;
-	case NRF_EINPROGRESS:
+	case NRF_EAI_SERVICE:
+		return DNS_EAI_SERVICE;
+	case NRF_EAI_SOCKTYPE:
+		return DNS_EAI_SOCKTYPE;
+	case NRF_EAI_INPROGRESS:
 		return DNS_EAI_INPROGRESS;
-	case NRF_ENETUNREACH:
-		errno = ENETUNREACH;
-	default:
+	case NRF_EAI_SYSTEM:
```

Comment:
Where will errno be set for EAI_SYSTEM?

Reply:
I think `errno` is already set by `nrf_getaddrinfo()` when it returns `NRF_EAI_SYSTEM`, so this should be enough?

Reply:
That's what I'd expect as well - @lemrey can you confirm that libmodem sets `errno` in this case?

Reply:
yep, `errno` is set by `libmodem` when `NRF_EAI_SYSTEM` is returned.